lua/orgmode/org/mappings.lua

```diff
@@ -218,37 +218,78 @@ function OrgMappings:handle_return(suffix)
   suffix = suffix or ''
   local item = Files.get_current_file():get_current_node()
   if item.type == 'headline' or item.node:parent():type() == 'headline' then
-    if item.type ~= 'headline' then
+    if item.type ~= 'stars' then
       ts_utils.goto_node(item.node:parent())
       item = Files.get_current_file():get_current_node()
     end
-    vim.fn.append(vim.fn.line('.'), { '', string.rep('*', item.level) .. ' ' .. suffix })
-    vim.fn.cursor(vim.fn.line('.') + 2, 0)
+    local section = item.node:parent():parent()
+    local end_row, _ = section:end_()
+    vim.api.nvim_buf_set_lines(0, end_row, end_row, false, { string.rep('*', item.level) .. ' ' .. suffix, '' })
+    vim.fn.cursor(end_row + 1, 0)
     return vim.cmd([[startinsert!]])
   end
-  if item.type == 'list' or item.type == 'listitem' then
-    if item.type == 'list' then
-      vim.fn.cursor(vim.fn.line('.'), vim.fn.col('$'))
-      item = Files.get_current_file():get_current_node()
+
+  if item.type == 'list' then
+    vim.cmd([[normal ^]])
```

Comment:
I think this should be `normal! ^`, since `normal` uses mapped key without `bang`. Also, does `startofline` option affect this functionality?

Reply:
good point about `!`
I think `startofline` should not affect this.